main.go

```diff
@@ -721,32 +740,34 @@ func (c *controller) saveHashring(ctx context.Context, hashring []receive.Hashri
 	return nil
 }
 
-func (c *controller) annotatePods(ctx context.Context, hashrings []receive.HashringConfig, statefulsets map[string]*appsv1.StatefulSet) {
-	for _, h := range hashrings {
-		if sts, exists := statefulsets[h.Hashring]; exists {
-			for i := 0; i < int(*sts.Spec.Replicas); i++ {
-				podName := fmt.Sprintf("%s-%d", sts.Name, i)
+func (c *controller) annotatePods(ctx context.Context) {
+	annotationKey := fmt.Sprintf("%s/%s", c.options.labelKey, "lastControllerUpdate")
 
-				pod, err := c.klient.CoreV1().Pods(c.options.namespace).Get(ctx, podName, metav1.GetOptions{})
-				if err != nil {
-					level.Error(c.logger).Log("msg", "failed to get pod", "err", err)
-				}
-
-				annotations := pod.Annotations
-				if annotations == nil {
-					annotations = make(map[string]string)
-				}
+	// Select pods that have a controllerLabel matching ours.
+	podList, err := c.klient.CoreV1().Pods(c.options.namespace).List(ctx,
```

Comment:
is this accurate? what if the controller is watching and acting on multiple StatefulSets? Won't it annotate Pods for each of those workloads even in the absence of a diff of their hashring which may be confusing?

Reply:
I don't believe the controller differentiates when it performs hashring updates as to which pods it touches. The sync code previously updated all pods on a hashring change, regardless of which hashring.